test fails on fresh clone without prior build

## Problem

Running `bun run test` on a fresh clone or new worktree fails because `packages/shared/dist/` does not exist.

```
Cannot find module '@agent-console/shared' or its corresponding type declarations.
```

## Root Cause

- `bun run test` runs `typecheck` first
- `typecheck` requires `packages/shared/dist/*.d.ts` (type declarations)
- `dist/` is created by `bun run build` but excluded from git via `.gitignore`
- On fresh clone/worktree, `dist/` doesn't exist

## Current Workaround

Run `bun run build` before `bun run test`:

```bash
bun install
bun run build  # Required!
bun run test
```

## Possible Solutions

1. **Add build to test script** - `"test": "bun run build && bun run typecheck && ..."` (slower)
2. **Use TypeScript project references** - Eliminates need for `dist/` but requires significant config changes
3. **Document the requirement** - Add note to README that `build` must run before `test`
4. **Keep as-is** - Developers typically run `dev` or `build` first anyway

## Notes

- This is NOT related to vitest→bun migration
- CI typically runs `build` before `test`, so this mainly affects local development
- `bun run dev` also builds shared, so developers who start with `dev` won't hit this

--- a/packages/client/package.json
+++ b/packages/client/package.json
@@ -5,7 +5,7 @@
   "type": "module",
   "scripts": {
     "dev": "vite",
-    "build": "tsc -b && vite build",
+    "build": "tsc --noEmit && vite build",
     "preview": "vite preview",
     "typecheck": "tsc --noEmit",
     "test": "bun test --preload ./src/test/setup.ts src/",

--- a/packages/client/tsconfig.json
+++ b/packages/client/tsconfig.json
@@ -17,8 +17,5 @@
     "noFallthroughCasesInSwitch": true,
     "noUncheckedSideEffectImports": true
   },
-  "include": ["src"],
-  "references": [
-    { "path": "../shared" }
-  ]
+  "include": ["src"]
 }

--- a/packages/server/tsconfig.json
+++ b/packages/server/tsconfig.json
@@ -6,8 +6,5 @@
     "lib": ["ES2022"],
     "types": ["bun", "node"]
   },
-  "include": ["src/**/*"],
-  "references": [
-    { "path": "../shared" }
-  ]
+  "include": ["src/**/*"]
 }

--- a/packages/shared/package.json
+++ b/packages/shared/package.json
@@ -3,12 +3,12 @@
   "version": "0.1.0",
   "private": true,
   "type": "module",
-  "main": "./dist/index.js",
-  "types": "./dist/index.d.ts",
+  "main": "./src/index.ts",
+  "types": "./src/index.ts",
   "exports": {
     ".": {
-      "types": "./dist/index.d.ts",
-      "import": "./dist/index.js"
+      "types": "./src/index.ts",
+      "import": "./src/index.ts"
     }
   },
   "scripts": {

--- a/packages/shared/tsconfig.json
+++ b/packages/shared/tsconfig.json
@@ -2,8 +2,7 @@
   "extends": "../../tsconfig.base.json",
   "compilerOptions": {
     "outDir": "./dist",
-    "rootDir": "./src",
-    "composite": true
+    "rootDir": "./src"
   },
   "include": ["src/**/*"]
 }

